stockflux-securities-master/src/components/Alert.js

```diff
@@ -0,0 +1,25 @@
+import React from "react";
+import "./Alert.css";
+import PropTypes from "prop-types";
+
+const Alert = ({ messages, type, className }) => {
+  return (
+    <ul className={`securities-alert-container ${className}`}>
+      {(Array.isArray(messages) ? messages : [messages]).map(
+        (message, index) => (
+          <li className={`securities-alert ${type}`} key={index}>
+            {message}
+          </li>
+        )
+      )}
+    </ul>
+  );
+};
+
+Alert.propTypes = {
+  messages: PropTypes.oneOfType([PropTypes.string, PropTypes.array]),
```

Comment:
Would be better to use this

    PropTypes.arrayOf(PropTypes.string)

Also, aren't messages and type both required?

Reply:
Okay, yeah will use that then. When I make type required it logs a warning on render because I'm setting formState to be null at the start. It might be worth leaving off the required for this filed then?

Reply:
Aren't you checking that the form state is one of the required states before you try to render it? If you're only allowing form state to be error or success then maybe you should only ever render the alert if the form state matches one of those instead?

Reply:
Yeah I wasn't checking that. I've added the check in and made the message proptype required now. 

There are no proptype warnings in the console any more